### ISSUE TITLE
Map.Entry.setValue equivalent for primitive mutable maps

### ISSUE THREAD

**Issue by victornoel** [eclipse/eclipse-collections#653]
Hi,

There is one feature that is great for performances in JDK's Map, it is the possibility to iterate on the `entrySet` and directly set the value from there.

With the primitive mutable maps of EC, you have to iterate on the keys then call `put`, which add some overhead.

**Comment by motlin** [comment]
Primitive maps have no entries or entry sets so this would hurt
performance. However, there could be an iteration pattern that takes a
function from current key and value to a new value and does an in place
mutation.

**Comment by victornoel** [comment]
@motlin yes, I wasn't thinking that we should implement it the same way as Map, something like your proposition seems good, maybe if @donraab has an opinion on the matter I could make a PR? :)

**Comment by donraab** [comment]
I agree with @motlin .  Implement an iteration pattern here.  Looking forward to seeing the PR. Thanks @victornoel !
